src/main/java/com/ecommerce/store/services/impl/ProductServiceImpl.java

```diff
@@ -34,20 +36,24 @@ public Product updateProduct(ProductUpdationDto productDto) {
         return repository
             .findById(productDto.getId())
             .map(existingProduct -> {
-                     if (StringUtils.hasLength(productDto.getName())) {
-                         existingProduct.setName(productDto.getName());
-                     }
-                     if (StringUtils.hasLength(productDto.getDescription())) {
-                         existingProduct.setDescription(productDto.getDescription());
-                     }
-                     if (productDto.getPrice() != null) {
-                         existingProduct.setPrice(productDto.getPrice());
-                     }
-                     return repository.save(existingProduct);
+                checkAllFieldExists(productDto, existingProduct);
+                return repository.save(existingProduct);
                  }
             ).orElseThrow(() -> new ProductNotExistsException("Product doesn't exist in the system to update"));
     }
 
+    private void checkAllFieldExists(ProductUpdationDto productDto, Product existingProduct) {
```

Comment:
@FuncGuy I think the function could be better-called `modifyUpdatedFields()`

Reply:
done!